### PR TITLE
Update Ubuntu version to 22

### PR DESCRIPTION
**Pull request by rmccar**
### What is the context of this PR?
- Updates and pins Ubuntu to version 22 for git actions

The Python version needed updating because of this and then because of that the pylint rules needed updating and then so did Black. So these changes were also made:
- Updates Python version to v3.10.8
- Updates the pylint rules for compatibility with Python 3.10
- Updates Black to v22.3.0
- Updates the PR template to standardise across repos

### How to review 
Checks pass

### Checklist
\*[ ] Tests updated

### CODE REVIEW

**Comment by rmccar** [comment]
[Upgrade from Ubuntu version 18.04 in GitHub Actions workflows](https://trello.com/c/UdVd2dtw/5531-upgrade-from-ubuntu-version-1804-in-github-actions-workflows)

File: Pipfile [hunk 1]
```diff
@@ -10,7 +10,7 @@ pyjwt = "*"
 requests = "*"
 
 [dev-packages]
-black = "==21.8b0"
+black = "==22.3.0"
```
Comment (Yuyuutsu) [inline Pipfile]: I think we should unpin the version for `black`

**Review by ONS-Guilhem-Forey — APPROVED** [review]
Lgtm

File: Pipfile (hunk 1 above)
Comment (rmccar) [reply Pipfile]: Unpinned and added details for why black needed updating to the description

**Review by Yuyuutsu — APPROVED** [review]
LGTM!

**Pull request merged**